File: solutions/ruby/run_solution.rb
require 'colorize'
require './migratory_birds'

#input parsing and reading
input00 = File.open("../../test-cases/input/input00.txt")
input05 = File.open("../../test-cases/input/input05.txt")

data = input00.map(&:chomp)
arr0 = data[1].split(" ").map(&:to_i)

data = input05.map(&:chomp)
arr5 = data[1].split(" ").map(&:to_i)


# problem application
start = Time.now
response00 = migratory_birds arr0
response05 = migratory_birds arr5
finish = Time.now

puts("solved all the test cases in " + (1000 * (finish - start)).round(2).to_s + "ms")

#expected outputs

output00File = File.open("../../test-cases/output/output00.txt")
output05File = File.open("../../test-cases/output/output05.txt")


data = output00File.map(&:chomp)
output00 = data[0].to_i

data = output05File.map(&:chomp)
output05 = data[0].to_i

if response00 === output00
	puts "INPUT 00".green
else
	puts "INPUT 00".red
end

if response05 === output05
	puts "INPUT 05".green
else
	puts "INPUT 05".red
end

File: solutions/java/RunSolution.java
import java.io.*; 
import java.util.Arrays;

public class RunSolution{

	public static final String ANSI_RED = "\u001B[31m";
	public static final String ANSI_GREEN = "\u001B[32m";
	public static final String ANSI_RESET = "\u001B[0m";

	public static void main(String[] args) throws Exception{

		//input parsing and reading
		File input00File = new File("../../test-cases/input/input00.txt");
		BufferedReader brInput00 = new BufferedReader(new FileReader(input00File));

		File input05File = new File("../../test-cases/input/input05.txt");
		BufferedReader brInput05 = new BufferedReader(new FileReader(input05File));


		String st;
		int currentLine = 0;
		String lineSplit[];

		int arr00[] = null;

		int arr05[] = null;

		while((st = brInput00.readLine()) != null){
			if(currentLine == 1){
				arr00 = Arrays
						.stream(st.split(" "))
						.mapToInt(Integer::parseInt)
						.toArray();

			}
			currentLine++;
		}

		currentLine = 0;

		while((st = brInput05.readLine()) != null){
			if(currentLine == 1){
				arr05 = Arrays
						.stream(st.split(" "))
						.mapToInt(Integer::parseInt)
						.toArray();
						
			}
			currentLine++;
		}

		// problem application

		int response00 = MigratoryBirds.run(arr00);
		int response05 = MigratoryBirds.run(arr05);
		

		// expected outputs

		File output00File = new File("../../test-cases/output/output00.txt");
		BufferedReader brOutput00 = new BufferedReader(new FileReader(output00File));

		File output05File = new File("../../test-cases/output/output05.txt");
		BufferedReader brOutput05 = new BufferedReader(new FileReader(output05File));


		int output00 = 0;
		int output05 = 0;

		while((st = brOutput00.readLine()) != null){
			output00 = Integer.parseInt(st.trim());
		}

		while((st = brOutput05.readLine()) != null){
			output05 = Integer.parseInt(st.trim());
		}

		System.out.println(((response00 == output00) ? ANSI_GREEN : ANSI_RED) + "INPUT 00");
		System.out.println(((response05 == output05) ? ANSI_GREEN : ANSI_RED) + "INPUT 05" + ANSI_RESET);
	}
}